sharedse/src/jvmTest/kotlin/com/ustadmobile/sharedse/network/containeruploader/TestContainerUploader.kt

```diff
@@ -0,0 +1,128 @@
+package com.ustadmobile.sharedse.network.containeruploader
+
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.spy
+import com.ustadmobile.core.account.Endpoint
+import com.ustadmobile.core.account.EndpointScope
+import com.ustadmobile.core.container.ContainerManager
+import com.ustadmobile.core.db.JobStatus
+import com.ustadmobile.core.db.UmAppDatabase
+import com.ustadmobile.lib.db.entities.Container
+import com.ustadmobile.lib.db.entities.ContainerUploadJob
+import com.ustadmobile.lib.util.sanitizeDbNameFromUrl
+import com.ustadmobile.port.sharedse.ext.generateConcatenatedFilesResponse
+import com.ustadmobile.port.sharedse.util.UmFileUtilSe
+import com.ustadmobile.sharedse.network.containeruploader.ContainerUploader.Companion.CHUNK_SIZE
+import com.ustadmobile.sharedse.network.NetworkManagerBle
+import com.ustadmobile.util.test.ext.bindNewSqliteDataSourceIfNotExisting
+import io.ktor.http.HttpStatusCode
+import kotlinx.coroutines.runBlocking
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.kodein.di.*
+import java.io.*
+import java.util.*
+import java.util.concurrent.TimeUnit
+import javax.naming.InitialContext
+
+class TestContainerUploader {
+
+    private lateinit var epubContainer: Container
+    private lateinit var entryListStr: String
+    private lateinit var containerManager: ContainerManager
+    private lateinit var appDb: UmAppDatabase
+    private lateinit var tmpFolder: File
```

Comment:
Use the Junit rule so this will be automatically deleted after the test:

```
@JvmField
@Rule
var tmpFolderRule = TemporaryFolder()
```

Then when you need a folder:
```
var tmpFolder = tmpFolderRule.newFolder()
```
... and the rule takes care of deleting it afterwards.